client/files_test.go

```diff
@@ -0,0 +1,27 @@
+// Copyright (c) 2022 Canonical Ltd
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License version 3 as
+// published by the Free Software Foundation.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+package client_test
+
+import (
+	"github.com/canonical/pebble/client"
+	. "gopkg.in/check.v1"
+)
+
+func (cs *clientSuite) TestListFiles(c *C) {
+	_, err := cs.cli.ListFiles(&client.ListFilesOptions{
+		Path: "/",
+	})
+	c.Assert(err, IsNil)
```

Comment:
Isn't this already tested on the daemon? https://github.com/canonical/pebble/blob/a755cf20f4baba6f81717d2c82e28518fa9e7926/internal/daemon/api_files_test.go#L100-L118

Also, I haven't found a way to test the client like this (setting up a Pebble API and spin up the client, instead of mocking the API responses)

Reply:
Fair enough. However, I think we want a bit more coverage of the conversion routines. For example, I think we should check at least some of the results here, and check that the handling of the pointers / optional fields like UserID/GroupID is working correctly.